main.tf

```diff
@@ -113,27 +119,56 @@ resource "aws_iam_role_policy_attachment" "existing_policies_for_eks_workers_rol
   role       = join("", aws_iam_role.default.*.name)
 }
 
+resource "aws_launch_template" "default" {
+  # We'll use this default if we aren't provided with a launch template during invocation
+  count = (local.enabled && (var.launch_template_id == null)) ? 1 : 0
+  block_device_mappings {
+    device_name = "/dev/xvda"
+
+    ebs {
+      volume_size = var.disk_size
+    }
+  }
+
+  instance_type = var.instance_type
+
+  dynamic "tag_specifications" {
+    for_each = ["instance", "volume", "elastic-gpu"]
+    content {
+      resource_type = tag_specifications.value
+      tags          = local.node_group_tags
+    }
+  }
+
+  # Override the default userdata if input is provided
+  # If none is provided, this value defaults to 'null', which defaults to the typical EKS node group userdata
```

Comment:
Will this clobber the default EKS node group userdata if provided? Is there anything important that runs via the default userdata or is it just always empty?

Reply:
The default userdata runs `bootstrap.sh` w/ the pertinent options to attach the node(s) to the EKS cluster. If you're using a custom launch template, or just a userdata override, you would have to include the entire script basically. 

And actually, I just realized that we have a solution in `terraform-aws-eks-workers` that will allow us to parameterize it, so let me add that and refactor this a little bit. Stay tuned!

Reply:
Sounds great

Reply:
All done if you want to have a look!